presto-spi/src/main/java/com/facebook/presto/spi/ClientRequestFilter.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public interface ClientRequestFilter
+{
+    List<String> getHeaderNames();
+
+    <T> Optional<Map<String, String>> getExtraHeaders(T additionalInfo);
```

Comment:
Why are we parameterizing `additionalInfo`?

Reply:
Since additionalInfo can vary in type, parameterizing it (<T>) is a good idea because it offers flexibility and reusability.

Reply:
Usually, plugins add a context object, and the runtime will populate the context object.

In this case, create a new `ClientRequestFilterContext`.  At first it has just one field, `Principal`, but it can be extended in the future.